File: NonUI_Others/server/TempDemo/app/src/main/java/ca/six/demo/actv_result/old/ActivityResultDemo.java
package ca.six.demo.actv_result.old;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import ca.six.demo.R;

/**
 * Created by songzhw on 2018/1/5.
 */
public class ActivityResultDemo extends AppCompatActivity implements View.OnClickListener {
    private TextView tvMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvMain = (TextView) findViewById(R.id.tvMain);
        tvMain.setText("Page B");
        tvMain.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        setResult(RESULT_OK);
        this.finish();
    }

}

File: Architecture/LearnArtitecture/app/src/main/java/ca/six/learn/artitecture/activity_decouple/result/ActivityResultDemo.kt
package ca.six.learn.artitecture.activity_decouple.result

import android.content.Intent
import android.os.Bundle
import android.view.MotionEvent
import android.widget.TextView
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
import androidx.appcompat.app.AppCompatActivity

class ActivityResultDemo : AppCompatActivity() {
    lateinit var resultLauncher: ActivityResultLauncher<Intent>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val tv = TextView(this)
        tv.textSize = 30f
        tv.text = "Page 1"
        setContentView(tv)

        // 自动生成requestCode, 所以不用我们操心了. (若有多种requestCode, 那就使用多个ActivityResultLauncher实例.
        // 算是彻底解决了if-else串这种不符合OC原则的做法了.
        resultLauncher = registerForActivityResult(StartActivityForResult()) { result: ActivityResult ->
                // ActivityResult类只有两个成员: int mResultCode, Intent mData
                val data = result.data
                println("szw code = ${ActivityResult.resultCodeToString(result.resultCode)}")
                println("szw result = ${data?.extras?.get("key")}") //=> code = RESULT_OK, result = value23
            }

    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        if (MotionEvent.ACTION_UP == event.action) {
            val it = Intent(this, ActivityResultB::class.java)
            resultLauncher.launch(it)
        }
        return super.onTouchEvent(event)
    }
}


